demos/vc_issuer/src/main.rs

```diff
@@ -0,0 +1,521 @@
+    verify_id_alias_credential_jws, AliasTuple,
+};
+
+/// We use restricted memory in order to ensure the separation between non-managed config memory (first page)
+/// and the managed memory for the archived data & indices.
+type Memory = RestrictedMemory<DefaultMemoryImpl>;
+type ConfigCell = StableCell<IssuerConfig, Memory>;
+
+const MINUTE_NS: u64 = 60 * 1_000_000_000;
+const CERTIFICATE_VALIDITY_PERIOD_NS: u64 = 5 * MINUTE_NS;
+const PROD_II_CANISTER_ID: &str = "rdmx6-jaaaa-aaaaa-aaadq-cai";
+
+thread_local! {
+    /// Static configuration of the archive set by init() or post_upgrade().
+    static CONFIG: RefCell<ConfigCell> = RefCell::new(ConfigCell::init(config_memory(), IssuerConfig::default()).expect("failed to initialize stable cell"));
+    static SIGNATURES : RefCell<SignatureMap> = RefCell::new(SignatureMap::default());
+    static EMPLOYEES : RefCell<HashSet<Principal>> = RefCell::new(HashSet::new());
+    static GRADUATES : RefCell<HashSet<Principal>> = RefCell::new(HashSet::new());
+}
+
+/// Reserve the first stable memory page for the configuration stable cell.
+fn config_memory() -> Memory {
+    RestrictedMemory::new(DefaultMemoryImpl::default(), 0..1)
+}
+
+#[cfg(target_arch = "wasm32")]
+use ic_cdk::println;
+
+#[derive(CandidType, Deserialize)]
+struct IssuerConfig {
+    /// Root of trust for checking canister signatures.
+    ic_root_key_raw: Vec<u8>,
+    /// List of canister ids that are allowed to provide id alias credentials.
+    idp_canister_ids: Vec<Principal>,
+}
+
+impl Storable for IssuerConfig {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        Cow::Owned(candid::encode_one(self).expect("failed to encode IssuerConfig"))
+    }
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        candid::decode_one(&bytes).expect("failed to decode IssuerConfig")
+    }
+    const BOUND: Bound = Bound::Unbounded;
+}
+
+impl Default for IssuerConfig {
+    fn default() -> Self {
+        Self {
+            ic_root_key_raw: extract_raw_root_pk_from_der(IC_ROOT_PK_DER)
+                .expect("failed to extract raw root pk from der"),
+            idp_canister_ids: vec![Principal::from_text(PROD_II_CANISTER_ID).unwrap()],
+        }
+    }
+}
+
+impl From<IssuerInit> for IssuerConfig {
+    fn from(init: IssuerInit) -> Self {
+        Self {
+            ic_root_key_raw: extract_raw_root_pk_from_der(&init.ic_root_key_der)
+                .expect("failed to extract raw root pk from der"),
+            idp_canister_ids: init.idp_canister_ids,
+        }
+    }
+}
+
+#[derive(CandidType, Deserialize)]
+struct IssuerInit {
+    /// Root of trust for checking canister signatures.
+    ic_root_key_der: Vec<u8>,
+    /// List of canister ids that are allowed to provide id alias credentials.
+    idp_canister_ids: Vec<Principal>,
+}
+
+#[init]
+#[candid_method(init)]
+fn init(init_arg: Option<IssuerInit>) {
+    let Some(init) = init_arg else {
+        // nothing to do
+        return;
+    };
+    apply_config(init);
```

Comment:
I think something like this might be more concise & will convey the intent better:

```rust
if let Some(init) = init_arg {
    apply_config(init)
}
```

Reply:
don't feel strongly about it!